WalletWasabi.Tests/UnitTests/UserInterfaceTest/PocketSelectionTests.cs

```diff
@@ -582,12 +582,72 @@ public void AutoSelectAllPockets()
 
 		selection.Reset(pockets.ToArray());
 
-		var output = selection.AutoSelectPockets();
+		var output = selection.AutoSelectPockets("");
 		Assert.Contains(pocket1, output);
 		Assert.Contains(pocket2, output);
 		Assert.Contains(pocket3, output);
 	}
 
+	[Fact]
+	public void AutoSelectOnlyKnownByRecipientPockets()
+	{
+		var selection = new LabelSelectionViewModel(Money.Parse("1.0"));
+
+		var pockets = new List<Pocket>();
+		pockets.AddPocket(0.8M, out var pocket1, CoinPocketHelper.PrivateFundsText);
+		pockets.AddPocket(1.1M, out var pocket2, "Dan");
+		pockets.AddPocket(0.5M, out var pocket3, CoinPocketHelper.UnlabelledFundsText);
+		pockets.AddPocket(1.1M, out var pocket4, "Dan", "Lucas");
+
+		selection.Reset(pockets.ToArray());
+
+		var output = selection.AutoSelectPockets("Dan");
+		Assert.DoesNotContain(pocket1, output);
+		Assert.Contains(pocket2, output);
+		Assert.DoesNotContain(pocket3, output);
+		Assert.DoesNotContain(pocket4, output);
+	}
+
+	[Fact]
+	public void AutoSelectOnlyKnownByAllMultipleRecipientPockets()
+	{
+		var selection = new LabelSelectionViewModel(Money.Parse("1.0"));
+
+		var pockets = new List<Pocket>();
+		pockets.AddPocket(0.8M, out var pocket1, CoinPocketHelper.PrivateFundsText);
+		pockets.AddPocket(1.1M, out var pocket2, "Dan");
```

Comment:
```suggestion
		pockets.AddPocket(1.1M, out var pocket2, "David");
```

By doing this the test does not pass but it should.

The problem is with this:
`var onlyKnownByRecipientPockets = knownByRecipientPockets.Where(x => x.Labels.All(label => recipient.Labels.Contains(label))).ToArray();`

We should only select coins that have the exact labels.

Reply:
Fixed, thanks